src/main/java/com/example/java/maven/cardGamePeterPan/Game.java

```diff
@@ -47,59 +48,86 @@ public void printPlayers() {
     }
 
     private void discardCardsIfPossible(Player player, int firstCardToDiscardIndex, int secondCardToDiscardIndex) {
-        if (player.areCardsMatching(firstCardToDiscardIndex, secondCardToDiscardIndex)) {
-            player.discardCardsFromHand(firstCardToDiscardIndex, secondCardToDiscardIndex);
-        }
-        else {
-            messagePrinter.printError(WRONG_CARDS_TO_DISCARD);
-            choseCardsToDiscard(player);
-        }
-    }
 
-    public int getValidCardIndex(Player player) {
-        int userInput = userInputProvider.getNumberOfChosenCard();
-        if (userInput > 0 && userInput <= player.getHand().size()) {
-            return (userInput - 1);
+        try {
+            if (player.areCardsMatching(firstCardToDiscardIndex, secondCardToDiscardIndex)) {
+                player.discardCardsFromHand(firstCardToDiscardIndex, secondCardToDiscardIndex);
+            }
+            else {
+                messagePrinter.printError(WRONG_CARDS_TO_DISCARD);
+                choseCardsToDiscard(player);
+            }
         }
-        else {
-            messagePrinter.printError(String.format("Your input must be an integer between 1 and %s", player.getHand().size()));
-            return getValidCardIndex(player);
+        catch (IndexOutOfBoundsException exception) {
+            messagePrinter.printError(exception.getMessage());
+            choseCardsToDiscard(player);
         }
     }
 
     public void getCardFromPreviousPlayerHand(Player player) {
         Player previousPlayer = getPreviousPlayer(player);
-        messagePrinter.printMessage(String.format("%s has %s cards. Which one would you like to draw?", previousPlayer.getName(), previousPlayer.getHand().size()));
-        player.takeCardFromAnotherPlayersHand(previousPlayer, getValidCardIndex(previousPlayer));
+        messagePrinter.printMessage(String.format("%s has %s cards. Which one would you like to draw?", previousPlayer.getName(), previousPlayer.getHandSize()));
+        try {
+            player.takeCardFromAnotherPlayersHand(previousPlayer, userInputProvider.getNumberOfChosenCard());
+        }
+        catch (IndexOutOfBoundsException exception) {
+            messagePrinter.printError(exception.getMessage());
+            getCardFromPreviousPlayerHand(player);
+        }
     }
 
-
     public Player getPreviousPlayer(Player player) {
-        int indexOfCurrentPlayer = playerList.indexOf(player);
+        List<Player> activePlayers = getActivePlayers();
+        int indexOfCurrentPlayer = activePlayers.indexOf(player);
         if ((indexOfCurrentPlayer - 1) < 0) {
-            return playerList.get(playerList.size() - 1);
+            return activePlayers.get(activePlayers.size() - 1);
         }
         else {
-            return playerList.get(indexOfCurrentPlayer-1);
+            return activePlayers.get(indexOfCurrentPlayer-1);
         }
     }
 
     public int countActivePlayers() {
```

Comment:
this method can be simplified to one line: `return getActivePlayers().size();`

Reply:
Done